table.go

```diff
@@ -77,6 +118,14 @@ func Diff(t1, t2 Table, diffRows bool) (*TableDiff, error) {
 	// For lookup.
 	keyMap := make(map[string]struct{}, len(key1))
 	for _, c := range key1 {
+		// Validate both tables have the key columns.
+		if _, ok := cols1[c]; !ok {
+			return fmt.Errorf("table 1 does not have key column `%s`", c)
+		}
+		if _, ok := cols2[c]; !ok {
```

Comment:
This checks if `cols2` has the `key1` columns in it. Shouldn't we instead be checking for the `key2` columns?

Reply:
Good observation! This is a bit deceiving and confusing, but there is only one `key` at least from the perspective of the `diff-table` program. However, programmatically you could technically create two tables with two different sets of keys. In theory, that could be fine if they have different names but still align. Obviously two sets of keys with different lengths would not work. I will update the code to perform this check and move this snippet into its own loop.